File: fix-all-remaining-conflicts.js
#!/usr/bin/env node

import fs from 'fs';
import path from 'path';
import { execSync } from 'child_process';

console.log('Starting comprehensive merge conflict resolution...');

// Function to find all files with merge conflicts
function findFilesWithConflicts() {
  try {
    // Find all TypeScript and JavaScript files
    const output = execSync('find . -name "*.ts" -o -name "*.tsx" -o -name "*.js" -o -name "*.jsx" | grep -v node_modules | grep -v .git', { encoding: 'utf8' });
    const files = output.trim().split('\n').filter(file => file.length > 0);
    
    const conflictedFiles = [];
    for (const file of files) {
      try {
        const content = fs.readFileSync(file, 'utf8');
        separatorFound = true;
        conflictType = 'separator';
        continue;
        inConflict = false;
        conflictType = '';
        separatorFound = false;
        continue;
      }
      
      if (inConflict) {
        if (conflictType === 'head') {
          // Skip HEAD content, we want remote
          continue;
        } else if (conflictType === 'separator') {
          // This is the separator, continue to collect remote content
          continue;
        } else if (separatorFound) {
          // This is remote content, keep it
          remoteContent.push(line);
        }
      } else {
        result.push(line);
      }
    }
    
    // Add any remaining remote content
    if (inConflict && remoteContent.length > 0) {
      result.push(...remoteContent);
    }
    
    fs.writeFileSync(filePath, result.join('\n'));
    return true;
  } catch (error) {
    console.error(`Error resolving conflicts in ${filePath}:`, error.message);
    return false;
  }
}

// Function to fix corrupted files by replacing with a basic template
function fixCorruptedFile(filePath) {
  try {
    const content = fs.readFileSync(filePath, 'utf8');
    
    // Check if file is severely corrupted (has many syntax errors)
    const lines = content.split('\n');
    let errorCount = 0;
    
    for (const line of lines) {

File: fix_all_remaining_conflicts.py
#!/usr/bin/env python3
"""
Fix all remaining merge conflicts in the project.
"""

import os
import glob
import re

def fix_merge_conflicts_in_file(file_path):
    """Fix merge conflicts in a single file."""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        if '<<<<<<< HEAD' not in content:
            return False
        
        print(f"Fixing merge conflicts in: {file_path}")
        
        # Split by merge conflict markers and keep the HEAD version
        lines = content.split('\n')
        resolved_lines = []
        in_conflict = False
        conflict_type = None
        
        for line in lines:
            if line.strip() == '<<<<<<< HEAD':
                in_conflict = True
                conflict_type = 'head'
                continue
            elif line.strip() == '=======':
                conflict_type = 'separator'
                continue
            elif line.strip().startswith('>>>>>>> '):
                in_conflict = False
                conflict_type = None
                continue
            elif line.strip().startswith('<<<<<<< ') and not line.strip().startswith('<<<<<<< HEAD'):
                in_conflict = True
                conflict_type = 'other'
                continue
            
            if not in_conflict or conflict_type == 'head':
                resolved_lines.append(line)
        
        resolved_content = '\n'.join(resolved_lines)
        
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(resolved_content)
        
        return True
    except Exception as e:
        print(f"Error fixing {file_path}: {e}")
        return False

def main():
    """Fix all merge conflicts in the project."""
    os.chdir('/workspace')
    
    # Find all files with merge conflicts
    patterns = ['**/*.tsx', '**/*.ts', '**/*.js', '**/*.jsx']
    all_files = []
    
    for pattern in patterns:
        for file_path in glob.glob(pattern, recursive=True):
            if 'node_modules' in file_path or '.git' in file_path:
                continue
            all_files.append(file_path)
    
    files_with_conflicts = []
    for file_path in all_files:
        try:
            with open(file_path, 'r', encoding='utf-8') as f:
                if '<<<<<<< HEAD' in f.read():
                    files_with_conflicts.append(file_path)
        except:
            continue
    
    print(f"Found {len(files_with_conflicts)} files with merge conflicts")
    
    # Fix each file
    fixed_count = 0
    for file_path in files_with_conflicts:
        if fix_merge_conflicts_in_file(file_path):
            fixed_count += 1
    
    print(f"Successfully fixed {fixed_count} files")

if __name__ == "__main__":
    main()